internal/util/pidlimit.go

```diff
@@ -27,32 +27,48 @@ import (
 )
 
 const (
-	procCgroup    = "/proc/self/cgroup"
-	sysPidsMaxFmt = "/sys/fs/cgroup/pids%s/pids.max"
+	procCgroup            = "/proc/self/cgroup"
+	sysPidsMaxFmtCgroupV1 = "/sys/fs/cgroup/pids%s/pids.max"
+	sysPidsMaxFmtCgroupV2 = "/sys/fs/cgroup%s/pids.max"
 )
 
 // return the cgouprs "pids.max" file of the current process
 //
-// find the line containing the pids group from the /proc/self/cgroup file
-// $ grep 'pids' /proc/self/cgroup
+// There are two different cgroup structures, cgroup v1 and v2
+//
+// For cgroup v1, find the line containing the pids group from the /proc/self/cgroup file
+// $ grep ':pids:' /proc/self/cgroup
 // 7:pids:/kubepods.slice/kubepods-besteffort.slice/....scope
 // $ cat /sys/fs/cgroup/pids + *.scope + /pids.max.
+//
+// The entry for cgroup v2 is always in the format "0::...scope", no subsystem given.
+// (see https://www.kernel.org/doc/Documentation/cgroup-v2.txt)
+
```

Comment:
nit: unneeded empty line. The function comments are not standard anyway (does not start the 1st line with `// getCgroupPidsFile ... `, so it does not matter much.

Reply:
Added function name and compacted comment.